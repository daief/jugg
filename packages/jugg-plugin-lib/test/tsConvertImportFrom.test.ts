import * as ts from 'typescript';
import transformerFactory from '../src/tsConvertImportFrom';

describe('ts-convert-import-from', () => {
  describe('convert-vue-2-js', () => {
    it('import, export, require', () => {
      const CODE = `
        import A from 'A/A.vue';
        import B from './A.vue';
        import C from 'A.vue';
        export * from '../../A.vue'
        export * from './A.tsx'
        export { A }
        require('A.vue');
        require('A.png');
        // not
        A.require('A.vue');

        ;(A, B, C)
      `;
      const tsResult = ts.transpileModule(CODE, {
        compilerOptions: {
          target: ts.ScriptTarget.ES2015,
        },
        transformers: {
          before: [transformerFactory()],
        },
      });
      const resultCode = tsResult.outputText;
      expect(resultCode).toMatchSnapshot();
    });

    it('with-comment-disable-convert', () => {
      const CODE = `
        import { C } from '../A.vue';  // jugg-lib-disable not converts

        export * from '../../A.vue' // jugg-lib-disable

        //  jugg-lib-disable
        export * from './B.vue'

        export * from './C.vue' // jugg-lib-disable

        require('A.vue'); // jugg-lib-disable

        // jugg-lib-disable
        require('A.vue');

        // jugg-lib-disable require('A.vue');

        require('A.vue'); /* jugg-lib-disable converts*/

        ;(C)
      `;
      const tsResult = ts.transpileModule(CODE, {
        compilerOptions: {
          target: ts.ScriptTarget.ES2015,
          lib: ['es6'],
          // module: ts.ModuleKind.CommonJS,
        },
        transformers: {
          before: [transformerFactory()],
        },
      });
      const resultCode = tsResult.outputText;
      expect(resultCode).toMatchSnapshot();
    });
  });
});
