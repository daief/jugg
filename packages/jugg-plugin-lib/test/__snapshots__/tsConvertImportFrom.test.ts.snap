// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ts-convert-import-from convert-vue-2-js import, export, require 1`] = `
"import A from \\"A/A.js\\";
import B from \\"./A.js\\";
import C from \\"A.js\\";
export * from \\"../../A.js\\";
export * from \\"./A.tsx\\";
export { A };
require(\\"A.js\\");
require(\\"A.png\\");
// not
A.require('A.vue');
;
(A, B, C);
"
`;

exports[`ts-convert-import-from convert-vue-2-js with-comment-disable-convert 1`] = `
"import { C } from '../A.vue'; // jugg-lib-disable not converts
export * from '../../A.vue'; // jugg-lib-disable
//  jugg-lib-disable
export * from \\"./B.js\\";
export * from './C.vue'; // jugg-lib-disable
require('A.vue'); // jugg-lib-disable
// jugg-lib-disable
require(\\"A.js\\");
// jugg-lib-disable require('A.vue');
require(\\"A.js\\"); /* jugg-lib-disable converts*/
;
(C);
"
`;
